frontend/src/components/file/FileUpload.vue

```diff
@@ -0,0 +1,94 @@
+<script setup lang="ts">
+import { storeToRefs } from 'pinia';
+import { ref } from 'vue';
+
+import { FileUpload } from '@/lib/primevue';
+import { documentService } from '@/services';
+import { useConfigStore } from '@/store';
+
+import type { FileUploadUploaderEvent } from 'primevue/fileupload';
+import type { Ref } from 'vue';
+
+// Props
+type Props = {
+  submissionId: string;
+};
+
+const props = withDefaults(defineProps<Props>(), {});
+
+const lastUploadedDocument = defineModel();
```

Comment:
Some kind of warning with this line:
> Prop "modelValue" should define at least its type

Maybe it's related to the defineProps thing not having a filled in type?